packages/esm-system-admin-app/src/change-password-modal/change-password-modal.component.tsx

```diff
@@ -0,0 +1,197 @@
+    (passwordInputValue, passwordInputFieldName) => {
+      if (passwordInputFieldName === 'newPassword') {
+        const uppercaseRegExp = /(?=.*?[A-Z])/;
+        const lowercaseRegExp = /(?=.*?[a-z])/;
+        const digitsRegExp = /(?=.*?[0-9])/;
+        const minLengthRegExp = /.{8,}/;
+        const passwordLength = passwordInputValue.length;
+        const uppercasePassword = uppercaseRegExp.test(passwordInputValue);
+        const lowercasePassword = lowercaseRegExp.test(passwordInputValue);
+        const digitsPassword = digitsRegExp.test(passwordInputValue);
+        const minLengthPassword = minLengthRegExp.test(passwordInputValue);
+        let errMsg = '';
+        if (passwordLength === 0) {
+          errMsg = t('passwordIsEmpty', 'Password is empty');
+        } else if (!uppercasePassword) {
+          errMsg = t('atLeastOneUppercase', 'At least one Uppercase');
+        } else if (!lowercasePassword) {
+          errMsg = t('atLeastOneLowercase', 'At least one Lowercase');
+        } else if (!digitsPassword) {
+          errMsg = t('atLeastOneDigit', 'At least one digit');
+        } else if (!minLengthPassword) {
+          errMsg = t('minimum8Characters', 'Minimum 8 characters');
+        } else if (passwordInput.oldPassword.length > 0 && passwordInput.newPassword === passwordInput.oldPassword) {
+          errMsg = t('newPasswordMustNotBeTheSameAsOld', 'New password must not be the same as password');
+        } else {
+          errMsg = '';
+          setIsNewPasswordInvalid(false);
+        }
+        setNewPasswordErr(errMsg);
+      } else if (
+        passwordInputFieldName === 'confirmPassword' ||
+        (passwordInputFieldName === 'newPassword' && passwordInput.confirmPassword.length > 0)
+      ) {
+        if (passwordInput.confirmPassword !== passwordInput.newPassword) {
+          setConfirmPasswordError(t('confirmPasswordMustBeTheSameAsNew', 'Confirm password must be the same as new'));
+        } else {
+          setConfirmPasswordError('');
+          setIsConfirmPasswordInvalid(false);
+        }
+      } else {
+        if (passwordInput.newPassword.length > 0 && passwordInput.newPassword === passwordInput.oldPassword) {
+          setOldPasswordErr(t('oldPasswordMustNotBeTheSameAsNew', 'Old password must not be the same as new'));
+        } else {
+          setOldPasswordErr('');
+          setIsOldPasswordInvalid(false);
+        }
+      }
+    },
+    [passwordInput.confirmPassword, passwordInput.newPassword, passwordInput.oldPassword, t],
+  );
+
+  useEffect(() => {
+    if (passwordInput.oldPassword !== '') {
+      handleValidation(passwordInput.oldPassword, 'oldPassword');
+    }
+    if (passwordInput.newPassword !== '') {
+      handleValidation(passwordInput.newPassword, 'newPassword');
+    }
+    if (passwordInput.confirmPassword !== '') {
+      handleValidation(passwordInput.confirmPassword, 'confirmPassword');
+    }
+  }, [handleValidation, passwordInput]);
```

Comment:
For the best client-side validation experience, we should probably use RHF and Zod here similar to how we're validating the Vitals and Biometrics form (and various other vanilla React forms in O3).